### PR TITLE
add Memory UT

### PR DESCRIPTION
**Pull request by iamyh**
add CRUD UT for MemoryDB

### CODE REVIEW

File: libdevcore/MemoryDB.cpp [hunk 1]
```diff
@@ -95,7 +96,7 @@ void MemoryDB::insert(h256 const& _h, bytesConstRef _v)
     else
         m_main[_h] = make_pair(_v.toString(), 1);
 #if ETH_PARANOIA
```
Comment (wheatli) [inline libdevcore/MemoryDB.cpp]: delete ETH_PARANOIA macro

File: libdevcore/MemoryDB.cpp [hunk 2]
```diff
@@ -95,7 +96,7 @@ void MemoryDB::insert(h256 const& _h, bytesConstRef _v)
...
-    LOG(DEBUG) << "INST" << _h << "=>" << m_main[_h].second;
+    LOG(DEBUG) << "INST " << _h << "=>" << m_main[_h].second;
```
Comment (wheatli) [inline libdevcore/MemoryDB.cpp]: It's better to leave a space for pretty show.

File: libdevcore/MemoryDB.cpp [hunk 3]
```diff
@@ -117,13 +118,13 @@ bool MemoryDB::kill(h256 const& _h)
             // If we get to this point, then there was probably a node in the level DB which we need
             // to remove and which we have previously used as part of the memory-based MemoryDB.
             // Nothing to be worried about *as long as the node exists in the DB*.
-            LOG(DEBUG) << "NOKILL-WAS" << _h;
+            LOG(DEBUG) << "NOKILL-WAS " << _h;
```
Comment (wheatli) [inline libdevcore/MemoryDB.cpp]: It's better to leave a space for pretty show.

File: libdevcore/MemoryDB.cpp [hunk 4]
```diff
@@ -117,13 +118,13 @@ bool MemoryDB::kill(h256 const& _h)
...
         }
-        LOG(DEBUG) << "KILL" << _h << "=>" << m_main[_h].second;
+        LOG(DEBUG) << "KILL " << _h << "=>" << m_main[_h].second;
```
Comment (wheatli) [inline libdevcore/MemoryDB.cpp]: It's better to leave a space for pretty show.

File: libdevcore/MemoryDB.cpp [hunk 5]
```diff
@@ -117,13 +118,13 @@ bool MemoryDB::kill(h256 const& _h)
...
     }
     else
     {
-        LOG(DEBUG) << "NOKILL" << _h;
+        LOG(DEBUG) << "NOKILL " << _h;
```
Comment (wheatli) [inline libdevcore/MemoryDB.cpp]: It's better to leave a space for pretty show.

File: libdevcore/MemoryDB.cpp (hunk 1 above)
Comment (iamyh) [reply libdevcore/MemoryDB.cpp]: ok.

**Pull request merged**